### PR TITLE
Fix no-thread-pool build on Windows

### PR DESCRIPTION
**Pull request by t8m**

### CODE REVIEW

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -23,6 +23,8 @@ jobs:
         platform:
           - arch: win64
             config: enable-fips
+          - arch: win64
```
Comment (Sashan) [inline .github/workflows/windows.yml:26]: I'm not fluent with workflow .yml file so I might be wrong. However it seems to me the newly added `arch win64` block is in conflict with existing `win64` block at line 24. Is that intentional?

File: crypto/thread/build.info [hunk 2]
```diff
@@ -14,7 +14,7 @@ ELSE
   IF[{- !$disabled{quic} -}]
     SOURCE[../../libssl]=$THREADS_ARCH
   ENDIF
-  $THREADS=api.c
+  $THREADS=api.c arch/thread_win.c
```
Comment (Sashan) [inline crypto/thread/build.info:17]: shall we also include `arch/thread_posix.c`? just question I have not tried fiddling with no-thread/no-thread-pool options yet.

thanks.

**Review by Sashan — CHANGES_REQUESTED** [review]

File: .github/workflows/windows.yml (hunk 1 above)
Comment (t8m) [reply .github/workflows/windows.yml:26]: No, it does not conflict.

File: crypto/thread/build.info (hunk 2 above)
Comment (t8m) [reply crypto/thread/build.info:17]: The rcu locks implementation does not depend on arch/thread_posix.c.

**Review by Sashan — APPROVED** [review]
thanks for clarification. I have no  further questions/

**Comment by t8m** [comment]
@nhorman this is alternative to #25357

**Review by nhorman — APPROVED** [review]

**Review by tom-cosgrove-arm — APPROVED** [review]

**Review by hlandau — APPROVED** [review]

**Review by shahsb — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master, 3.4, 3.3 and 3.2 branches. Thank you for the reviews.

**Pull request closed without merging**